os_dep.c

```diff
@@ -2387,7 +2387,14 @@ GC_unix_mmap_get_mem(size_t bytes)
     return GC_unix_mmap_get_mem(bytes);
   }
 #      endif
+
+#      ifndef CHERI_PURECAP
   last_addr = PTR_ALIGN_UP((ptr_t)result + bytes, GC_page_size);
+#      else
+  /* CHERI_PURECAP: Non-NULL addresses don't work in CheriABI */
+  last_addr = NULL;
```

Comment:
I've already fixed this in a different way - use address for the hint instead of the pointer (with capabilities): https://github.com/ivmai/bdwgc/commit/5fe8a391eb3d7c13a44297806734dff6989dc58d